pool/client.go

```diff
@@ -631,7 +631,7 @@ func (c *Client) handleSubmitWorkRequest(ctx context.Context, req *Request, allo
 		c.sendMessage(resp)
 		return err
 	}
-	log.Tracef("Work %s accepted by the network", hash)
+	log.Tracef("Work %s accepted by the network", header.BlockHash())
```

Comment:
Calculating block hashes isn't all that expensive, but it seems wasteful to calculate them back to back like this.

Maybe?

```Go
blockHash := header.BlockHash()
work := NewAcceptedWork(blockHash.String(), ...)
...
log.Tracef("Work %s accepted by the network", blockHash)
```

Reply:
Multiple other log lines are writing the powHash instead of the block hash, so to maintain consistency I've reverted the logging change.

Reply:
Alright.  I'll approve since it's more important to fix the issue for now.

That said, I think there is a distinct difference though that the code is not really properly making in here that will make it very difficult to track what is going on in the future from the logs.

Namely, shares submitted by workers should be based on the proof of work hash.  Meanwhile, _blocks_ submitted to the network should be based on the block hash.

The reason for that is that there is no way to look up a block by its proof of work hash, because it is not intended to be an identifier (that is what the block hash is for).

Given that, I think there really should be a logged mapping of the block hash <-> proof of work hash for blocks submitted to the network.  Otherwise, as mentioned, admins won't be able to easily correlate and track down issues.